spec/components/rails_component_spec.rb

```diff
@@ -38,53 +38,55 @@
     { context: }
   end
 
-  let(:test_cell) do
-    Class.new(described_class) do
-      property :foo
+  class TestComponent < described_class
+    include ViewComponent::InlineTemplate
 
-      def link
-        link_to "<strong>HTML</strong>", '/foo/bar'
-      end
+    property :foo
+
+    erb_template <<~ERB
+      <%= foo %>
+    ERB
+
+    def link
+      link_to "<strong>HTML</strong>", '/foo/bar'
+    end
 
-      def content
-        content_tag(:div) do
-          content_tag(:span) do
-            "<script>foo</script>"
-          end
+    def content
+      content_tag(:div) do
+        content_tag(:span) do
+          "<script>foo</script>"
         end
       end
     end
+
   end
 
-  let(:instance) { test_cell.new model, options }
+  let(:component) { TestComponent.new(model, **options) }
 
   before do
     allow(controller).to receive(:view_context).and_return(action_view)
   end
 
   shared_examples 'uses action_view helpers' do
     describe '#link' do
-      subject { instance.link }
-
+      subject { component.link }
       it 'uses link_to from rails with escaping' do
-        expect(subject.to_s).to eq %(<a href="/foo/bar">&lt;strong&gt;HTML&lt;/strong&gt;</a>)
+        expect(subject).to eq %(<a href="/foo/bar">&lt;strong&gt;HTML&lt;/strong&gt;</a>)
       end
     end
 
     describe '#foo' do
-      subject { instance.foo }
-
+      subject { component.call }
       it 'escapes the property' do
-        expect(subject.to_s).to eq "&lt;strong&gt;Some HTML here!&lt;/strong&gt;"
+        render_inline component
```

Comment:
Dear reviewer, calling the `#foo` method directly will not do any escaping, but rendering the component and including the value of `#foo` makes it escaped. However there is also a gotcha that I'm not sure if it is an intended behaviour.
Writing the test the following way fails to escape:

```ruby
expect(page).to have_selector("body", text: "&lt;strong&gt;Some HTML here!&lt;/strong&gt;")
```
But calling the `component.call` (which is essentially is being called by render as well), will output the escaped html string. I may be overlooking something, please double check.